src/utilities/selection/SelectionZone.tsx

```diff
@@ -196,6 +198,7 @@ export class SelectionZone extends React.Component<ISelectionZoneProps, any> {
             selection.setAllSelected(false);
           }
           selection.setIndexSelected(index, !isSelected, true);
+          onItemSelected(selection.getItems()[index], index, !isSelected);
```

Comment:
i don't think this is right.

selection object itself is something that can be externallymanipulated outside of selection zone. The caller should use the selection object to get selection changes. 

A better solution here would be: expose the selection object publicly via getter, make it part of the IDetailsList contract. Then make sure the selection object has the hooks you need to observe changes.

Consider this; i have a MarqueeSelection component coming. It takes in a selection object, and enables you to drag a box around items that are selected. When it wants to set something selected, it interacts with Selection, not SelectionZone. It will say selection.setSelected(index, true) on something. Then a change event should fire and the consumer should deal with that.